src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsEventPublisher.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.ApplicationInsights;
+using System.Diagnostics;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+{
+    /// <summary>
+    /// Listens to <see cref="Activity"/> events from feature management and sends them to Application Insights.
+    /// </summary>
+    internal class ApplicationInsightsEventPublisher : IDisposable
+    {
+        private readonly TelemetryClient _telemetryClient;
+        private readonly ActivityListener activityListener;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationInsightsEventPublisher"/> class.
+        /// </summary>
+        /// <param name="telemetryClient">The Application Insights telemetry client.</param>
+        public ApplicationInsightsEventPublisher(TelemetryClient telemetryClient)
+        {
+            _telemetryClient = telemetryClient ?? throw new ArgumentNullException(nameof(telemetryClient));
+
+            activityListener = new ActivityListener
+            {
+                ShouldListenTo = (activitySource) => activitySource.Name == "Microsoft.FeatureManagement",
```

Comment:
Good callout. The problem to me is that they're currently in different packages. One string is in the primary `Microsoft.FeatureManagement` while the other is in `Microsoft.FeatureManagement.Telemetry.ApplicationInsights`. I could expose it as a public or InternalsVisibleToAttribute to share between the packages but it didn't feel valuable enough.

Reply:
`internal` + `InternalsVisibleTo` would be the best in this case indeed to avoid overexposing it. I see it is a bit convoluted though still, probably not worth it as you said.